src/command.ts

```diff
@@ -326,6 +347,75 @@ export class Command<C extends Context = Context> implements MiddlewareObj<C> {
     return this;
   }
 
+  /**
+   * Finds the matching command in the given context
+   *
+   * @example
+   * ```ts
+   * // ctx.msg.text = "/delete_123 something"
+   * const match = Command.findMatchingCommand(/delete_(.*)/, { prefix: "/", ignoreCase: true }, ctx)
+   * // match is { command: /delete_(.*)/, rest: ["something"], match: ["delete_123"] }
+   * ```
+   */
+  public static findMatchingCommand(
+    command: MaybeArray<string | RegExp>,
+    options: CommandOptions,
+    ctx: Context,
+  ): CommandMatch | null {
+    const { matchOnlyAtStart, prefix, targetedCommands } = options;
+
+    if (!ctx.has(":text")) return null;
+
+    if (matchOnlyAtStart && !ctx.msg.text.startsWith(prefix)) {
+      return null;
+    }
+
+    const commandNames = ensureArray(command);
+    const escapedPrefix = prefix.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+    const commandRegex = new RegExp(
+      `${escapedPrefix}(?<command>[^@ ]+)(?:@(?<username>[^\\s]*))?(?<rest>.*)`,
```

Comment:
I would like to add [super-expressive](https://github.com/francisrstokes/super-expressive) as a dep, since we are using quite a few complex regexs across the plugin

Reply:
I agree, but I'd rather move on with this and add the dep on a new PQ